src/Security/Authentication/Facebook/src/FacebookHandler.cs

```diff
@@ -60,16 +58,15 @@ protected override async Task<AuthenticationTicket> CreateTicketAsync(ClaimsIden
 
         private string GenerateAppSecretProof(string accessToken)
         {
-            using (var algorithm = new HMACSHA256(Encoding.ASCII.GetBytes(Options.AppSecret)))
+            var key = Encoding.ASCII.GetBytes(Options.AppSecret);
+            var tokenBytes = Encoding.ASCII.GetBytes(accessToken);
+            var hash = HMACSHA256.HashData(key, tokenBytes);
+            var builder = new StringBuilder();
```

Comment:
I considered changing this to use `Convert.ToHexString().ToLowerInvariant()` too, but wasn't sure on which was "better" on a non-test code path given simplifying it would require the `ToLower*()` call. Will change if the simplicity would be preferred.

Reply:
You might do all this without any allocation if you want:
* Stackalloc for the hash signature
* Use the Span based API to get the hex encoding on a span
* Use ToLowerInvariant to convert ToLower
* Create the final string from the span

 https://docs.microsoft.com/en-us/dotnet/api/system.memoryextensions.tolowerinvariant?view=net-5.0

Reply:
Like this?

```csharp
var secretBytes = Encoding.ASCII.GetBytes(Options.AppSecret);
var tokenBytes = Encoding.ASCII.GetBytes(accessToken);

Span<byte> hashedBytes = stackalloc byte[32];

var written = HMACSHA256.HashData(secretBytes, tokenBytes, hashedBytes);

if (written != hashedBytes.Length)
{
    throw new InvalidOperationException("Could not compute the hash for the 'appsecret_proof' value.");
}

var hashStringUpper = Convert.ToHexString(hashedBytes).AsSpan();

Span<char> hashStringLower = stackalloc char[hashStringUpper.Length];

if (hashStringUpper.ToLowerInvariant(hashStringLower) != hashStringUpper.Length)
{
    throw new InvalidOperationException("Could not compute the hash for the 'appsecret_proof' value.");
}

return new(hashStringLower);
```

Reply:
Ah, my bad. I saw the ToHexString span API and assumed it could do it without the intermediate hex string conversion. I'm surprised that there is no overload to do this without generating a string

Reply:
I thought it was ToHexString(span, resultSpan)

Reply:
Filed an issue on the runtime https://github.com/dotnet/runtime/issues/58937 to see if it gets added in the future